### PR TITLE
[Icds Dashboard] Add primary school form ucr

### PR DESCRIPTION
**Pull request by Rohit25negi**
Creating new UCR for groud work for this: https://app.asana.com/0/1112385193248823/1138961090831547.

Summary: We want to exclude some children from pse_eligible if he is going to primary/private school. This information comes from this form for which this UCR is to be created.

I would like If I could deploy it by Monday so that I could do the datafilling

### CODE REVIEW

File: custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        "expression": {
+          "type": "ancestor_location",
+          "location_id": {
+            "type": "named",
+            "name": "user_location_id"
+          },
+          "location_type": "state",
+          "location_property": "_id"
+        },
+        "create_index": true
```
Comment (emord) [inline custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json]: this index isn't needed as its covered by `(state_id, timeend)`

File: custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      "type": "coalesce",
+      "expression": {
+        "type": "property_path",
+        "property_path": ["form", "case_load_person_0", "case", "@case_id"]
+      },
+      "default_expression": {
+        "type": "constant",
+        "constant": [0]
+      }
+    },
```
Comment (emord) [inline custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json]: Why does this need a base_item_expression? from the example XML, there's only one case that could be in the form and it doesn't appear that this data source ever references the iteration number. I think that this and the `repeat_iteration` in the primary key could be dropped

File: custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json]: Ow. I did not know that. removed!

File: custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json (hunk 2 above)
Comment (Rohit25negi) [reply custom/icds_reports/ucr/data_sources/dashboard/primary_private_school_form_ucr.json]: Yeah, that makes sense, and that's why repeat_iteration will be always 0. CLeaned the UCR!

**Pull request merged**